ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateUtil.java

```diff
@@ -0,0 +1,133 @@
+     */
+    @NonNull
+    String convertDateFromServerToWidgetFormat(@Nullable final String serverDate) {
+        if (isServerDateNotMatched(serverDate)) {
+            return "";
+        }
+        try {
+            final Calendar calendar = Calendar.getInstance();
+            calendar.setTime(mServerDateFormat.parse(serverDate));
+            return mWidgetDateFormat.format(calendar.getTime());
+        } catch (NumberFormatException | ParseException | IndexOutOfBoundsException e) {
+            Log.e(TAG, e.getMessage());
+            return "";
+        }
+    }
+
+    /**
+     * Build Date in the server format (yyyy-MM-dd)
+     *
+     * @param year       the selected year
+     * @param month      the selected month (0-11 for compatibility with {@link Calendar#MONTH})
+     * @param dayOfMonth th selected day of the month (1-31, depending on month)
+     * @return String date in the format (yyyy-MM-dd)
+     */
+    @NonNull
+    String buildDateFromDateDialogToServerFormat(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = getCalendar(year, month, dayOfMonth);
+        return mServerDateFormat.format(calendar.getTime());
+    }
+
+    /**
+     * Build Date in the DateWidget format (dd-MMM-yyyy)
+     *
+     * @param year       the selected year
+     * @param month      the selected month (0-11 for compatibility with {@link Calendar#MONTH})
+     * @param dayOfMonth th selected day of the month (1-31, depending on month)
+     * @return String date in the format (yyyy-MM-dd)
+     */
+    @NonNull
+    String buildDateFromDateDialogToWidgetFormat(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = getCalendar(year, month, dayOfMonth);
+        return mWidgetDateFormat.format(calendar.getTime());
+    }
+
+    /**
+     * Convert Date from server format (yyyy-MM-dd) to Calendar
+     *
+     * @param serverDate Date from server
+     * @return Calendar with date from server
+     */
+    @NonNull
+    Calendar convertDateFromServerFormatToCalendar(@Nullable final String serverDate) {
+        final Calendar calendar = Calendar.getInstance();
+        if (isServerDateNotMatched(serverDate)) {
+            return calendar;
+        }
+
+        try {
+            calendar.setTime(mServerDateFormat.parse(serverDate));
+        } catch (ParseException e) {
+            Log.e(TAG, e.getMessage());
+        }
+        return calendar;
+    }
+
+    @NonNull
+    private Calendar getCalendar(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(year, month, dayOfMonth, 0, 0, 0);
+        return calendar;
+    }
+
+    private boolean isServerDateNotMatched(@Nullable final String serverDate) {
+        if (serverDate == null) {
+            return true;
+        }
+        final Matcher matcher = Pattern.compile(DATE_PATTERN).matcher(serverDate);
```

Comment:
you can do like the ff code:
```
    private boolean isServerDateValid(@Nullable final String serverDate) {
        Date date = null;
        if (!TextUtils.isEmpty(serverDate)) {
            SimpleDateFormat formatter = new SimpleDateFormat(SERVER_DATE_PATTERN, Locale.getDefault());
            formatter.setLenient(Boolean.FALSE); // make it strict
            try {
                date = formatter.parse(serverDate);
            } catch (ParseException e) {
                date = null;
            }
        }

        return date != null;
    }
```
This code can take care of format, valid months, days, and year, plus it can take care of leap year computation, no need to duplicate